cmd/erasure-multipart.go

```diff
@@ -1027,7 +1027,7 @@ func (er erasureObjects) CompleteMultipartUpload(ctx context.Context, bucket str
 	fi.IsLatest = true
 
 	// Success, return object info.
-	return fi.ToObjectInfo(bucket, object), nil
+	return fi.ToObjectInfo(bucket, object, opts.Versioned || opts.VersionSuspended), nil
```

Comment:
opts.Versioned and opts.VersionSuspended are not guaranteed to be set correctly in all cases - for e.g in completeMultipartOpts and other cases where ObjectOptions is defined on the fly. It may be more reliable to check globalBucketVersioningSys instead within fi.ToObjectInfo?

Reply:
No, we shouldn't be calling in the potential for disk I/O -  opts.Versioned or Suspended will always be set correctly from this PR onwards - that is the crux of this change.  CompleteMultipartUpload definitely sets the correct value however that is not needed because the required state is already set during newMultipartUpload(). 

If we cannot ascertain the state of the bucket we will panic such API calls @poornas